uefi/firmwarefile.go

```diff
@@ -41,16 +41,42 @@ const (
 const (
 	// FileHeaderMinLength is the minimum length of a firmware file header.
 	FileHeaderMinLength = 0x18
+	// FileHeaderExtMinLength is the minimum length of an extended firmware file header.
+	FileHeaderExtMinLength       = 0x20
+	emptyBodyChecksum      uint8 = 0xAA
 )
 
+// IntegrityCheck holds the two 8 bit checksums for the file header and body separately.
+type IntegrityCheck struct {
+	Header uint8
+	File   uint8
+}
+
+type fileAttr uint8
+
 // FirmwareFileHeader represents an EFI File header.
 type FirmwareFileHeader struct {
-	Name           uuid.UUID // This is the GUID of the file.
-	IntegrityCheck [2]uint8
-	Type           FVFileType
-	Attributes     uint8
-	Size           [3]uint8
-	State          uint8
+	Name       uuid.UUID // This is the GUID of the file.
+	Checksum   IntegrityCheck
+	Type       FVFileType
+	Attributes fileAttr
+	Size       [3]uint8
+	State      uint8
+}
+
+// Checks if the large file attribute is set
+func (a fileAttr) isLarge() bool {
+	return a&0x01 != 0
+}
+
+// Checks if we need to checksum the file body
+func (a fileAttr) checksum() bool {
```

Comment:
the name `checksum` is a bit counter-intuitive, it hints to a method that will return a checksum rather than a method that will return whether the checksum is set and has to be verified. What about renaming it to `hasChecksum` or similar?